packages/lodestar/src/executionEngine/interface.ts

```diff
@@ -63,13 +64,19 @@ export type ApiPayloadAttributes = {
   /** DATA, 20 Bytes - suggested value for the coinbase field of the new payload */
   suggestedFeeRecipient: DATA;
 };
+
+export type ProposerPreparationData = {
+  validatorIndex: ValidatorIndex;
+  feeRecipient: ExecutionAddress;
+};
 /**
  * Execution engine represents an abstract protocol to interact with execution clients. Potential transports include:
  * - JSON RPC over network
  * - IPC
  * - Integrated code into the same binary
  */
 export interface IExecutionEngine {
+  proposers: MapDef<ValidatorIndex, {epoch: Epoch; feeRecipient: ExecutionAddress}>;
```

Comment:
Is it really the best place to put this data here? This data is agnostic of transport, and ExecutionEngineDisabled having to be aware of it shows it's maybe not the best place. If this map is unique to each beacon node, why not keep in the BeaconChain class? Does block production have access to it?